### ISSUE TITLE
Documentation for special values of key in each template helper 

### ISSUE THREAD

**Issue by saleswhale-vincent** [emberjs/ember.js#16719]
Hello! I'm not sure if this is the right place for this but I wasn't sure where to ask. In the [current documentation for the each template helper](https://emberjs.com/api/ember/3.2/classes/Ember.Templates.helpers/methods/each?anchor=each) it talks about specifying keys but makes no mention of the special values for key, namely: 

@index - The index of the item in the array.
@item - The item in the array itself. This can only be used for arrays of strings or numbers.
@guid - Generate a unique identifier for each object (uses Ember.guidFor). 

Which I discovered in this [comment](https://discuss.emberjs.com/t/new-glimmer-consideration-each-key/8128/9)

Are these special values actually still valid in 3.2? And if they are, is it alright if I write them back into the documentation?

**Comment by rwjblue** [comment]
Correct. The following should be considered public API’s for the specific helpers.

`{{each`:

* `@index`
* `@identity`

`{{each-in`:

* `@key`
* `@index`
* `@identity`

Tweaks to the API docs for these helpers to add mention of the available special values seems good.

**Comment by rwjblue** [comment]
Also, thought I should cross link this RFC: https://github.com/emberjs/rfcs/pull/321

**Comment by Serabe** [comment]
Thanks, applying labels accordingly.

**Comment by pixelhandler** [comment]
@emberjs/learning-team-managers perhaps we can update the API docs accordingly

**Comment by kategengler** [comment]
Looks like we've documented for `each` but not yet for `each-in`

**Comment by johanrd** [comment]
@kategengler thanks, good catch, an attempt to pull it over the finishing line here: https://github.com/emberjs/ember.js/pull/21536
